msg/templates/urtps/Subscriber.cpp.em

```diff
@@ -131,28 +133,16 @@ void @(topic)_Subscriber::SubListener::onSubscriptionMatched(Subscriber* sub, Ma
 void @(topic)_Subscriber::SubListener::onNewDataMessage(Subscriber* sub)
 {
         // Take data
-@[if 1.5 <= fastrtpsgen_version <= 1.7]@
-@[    if ros2_distro]@
-        @(package)::msg::dds_::@(topic)_ st;
-@[    else]@
-        @(topic)_ st;
-@[    end if]@
-@[else]@
-@[    if ros2_distro]@
-        @(package)::msg::@(topic) st;
-@[    else]@
-        @(topic) st;
-@[    end if]@
-@[end if]@
-
-        if(sub->takeNextData(&st, &m_info))
+        if(sub->takeNextData(&msg, &m_info))
         {
             if(m_info.sampleKind == ALIVE)
             {
                 // Print your structure data here.
                 ++n_msg;
                 //std::cout << "Sample received, count=" << n_msg << std::endl;
                 has_msg = true;
+                
+                cv_msg->notify_all();
```

Comment:
You're right, it's kinda solved by using the wait_for instead of wait in the t_send worker thread. Thus the only thing notify does is to ensure the t_send thread gets resumed on time. Ideally you would get rid of the whole hasMsg scheme and make a concurrent queue which would be fully synchronized. 

An first attempt to make it fully thread safe is here https://github.com/PetervdPerk-NXP/px4_ros_com/commit/51cac5b8c00384e6bdaef7ad8920d4c4feefaab8, however it needs more intensive testing and cleanup. But feel free to provide some feedback on the queue-based code.